### PR TITLE
Prepare 3.0.0 release

### PR DESCRIPTION
**Pull request by Autophagy**

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/index.rst [hunk 1]
```diff
@@ -17,6 +17,17 @@ information and changelog.
 .. _the CrateDB download page: https://crate.io/download/
 .. _the release archives: https://cdn.crate.io/downloads/releases/
 
+3.x
+===
+
+3.0.x
+=====
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/index.rst]: This headline should be a sub-headline of 3.x, so only single underline.

File: blackbox/docs/appendices/release-notes/3.0.0.rst [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+- The gateway recovery table setting ``recovery.initial_shards`` has been
+  removed. Nodes will recover their unassigned local primary shards
+  immediatelly after restart.
+
+- The ``_node`` column on ``sys.shards`` and ``sys.operations`` has been
+  renamed to ``node``, is now visible by default and has been trimmed to only
+  include ``node['id']`` and ``node['name']``. In order to get all information
+  a join query can be used with ``sys.nodes``.
+
+- ``discovery.type`` setting has been removed. To enable EC2 discovery
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/3.0.0.rst]: There are three points about removed settings, but all are phrased slightly different. Maybe we could unify the wording, e.g.

* The store level throttling settings ``indices.store.throttle.*`` have been removed.
* The gateway table setting `recovery.initial_shards` has been removed. ...
* The discovery setting `discovery.type` has been removed. ...

File: blackbox/docs/appendices/release-notes/3.0.0.rst [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+- Added support for ``DEALLOCATE`` statement which is used by certain Postgres
+  Wire Protocol clients (e.g. libpq) to deallocate a prepared statement and
+  release its resources.
+
+- Implemented ``constraint_schema`` and ``table_schema`` in
+  ``information_schema.key_column_usage`` correctly and documented the full
+  table schema.
+
+- Implemented the ``Block Hash Join`` algorithm which is now used for
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/3.0.0.rst]: I don't think the name of the algorithm should be formatted like inline code. Same for "Equi-Joins". They are just regular terms.

File: blackbox/docs/appendices/release-notes/3.0.0.rst [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+- The ``_node`` column on ``sys.shards`` and ``sys.operations`` has been
+  renamed to ``node``, is now visible by default and has been trimmed to only
+  include ``node['id']`` and ``node['name']``. In order to get all information
+  a join query can be used with ``sys.nodes``.
+
+- ``discovery.type`` setting has been removed. To enable EC2 discovery
+  ``discovery.zen.hosts_provider`` setting needs to be set to ``ec2``.
+
+Changes
+-------
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/3.0.0.rst]: Would it makes sense to order the changes so they are roughly grouped by topic? E.g.: dependency updates, new features by feature, general improvements, removed stuff.

**Comment by Autophagy** [comment]
@chaudum Pushed a fixup. I've unified the language about settings removal/renaming. I've also re-ordered the breaking changes a little bit, and grouped the changes section by:

- ES Changes
- Tooling changes (admin-ui/crash/etc)
- Postgres features
- New features
- Changes
- Fixes
- Docs

Let me know if you think this is appropriate.

File: blackbox/docs/appendices/release-notes/3.0.0.rst [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+- DEPRECATED: The ``ON DUPLICATE KEY UPDATE`` clause has been deprecated in
+  favor of the ``ON CONFLICT DO UPDATE SET`` clause.
+
+- Implemented the Block Hash Join algorithm which is now used for Equi-Joins.
+
+- Added new ``sys.health`` system information table to expose the health of all
+  tables and table partitions.
+
+- Added a new ``bengali`` language analyzer and a ``bengali_normalization``
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/3.0.0.rst]: maybe move one item below so fulltext changes are grouped

File: blackbox/docs/appendices/release-notes/3.0.0.rst [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+  ``information_schema.key_column_usage`` correctly and documented the full
+  table schema.
+
+- Statistics for jobs and operations are enabled by default. If you don't need
+  any statistics, please set ``stats.enabled`` to ``false``.
+
+- Changed ``BEGIN`` and ``SET SESSION`` to no longer require ``DQL``
+  permissions on the ``CLUSTER`` level.
+
+- Added 'epoch' argument to the extract function which returns the number of
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/3.0.0.rst]: `epoch` could be seen as a keyword. inline code formatting would be appropriate.

**Pull request merged**